compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/SwingComposeSceneLayer.desktop.kt

```diff
@@ -149,10 +141,17 @@ internal class SwingComposeSceneLayer(
         containerSize = IntSize(windowContainer.width, windowContainer.height)
     }
 
+    private fun onDrawRectChange(rect: Rect) {
+        val bounds = rect.roundToIntRect().translate(drawBounds.topLeft)
+        drawBounds = bounds
+        setComponentBounds(bounds)
+    }
+
     private fun createSkiaLayerComponent(mediator: ComposeSceneMediator): SkiaLayerComponent {
+        val skikoView = RecordDrawRectSkikoViewDecorator(mediator, ::onDrawRectChange)
         return SwingSkiaLayerComponent(
```

Comment:
This looks like a bug because each subsequent `drawBounds` value appears to depend directly on the previous one. What is hidden here is that `rect` is relative to the canvas, which is itself offset each time by `-drawBounds.topLeft`, so it sums up to 0.

I suggest this instead, which makes the dependency clearer:
```
    private fun onDrawRectChange(rect: Rect) {
        rect.roundToIntRect().let {
            drawBounds = it
            setDialogBounds(it)
        }
    }

    private fun createSkiaLayerComponent(mediator: ComposeSceneMediator): SkiaLayerComponent {
        val skikoView = OverlaySkikoViewDecorator(
            RecordDrawRectSkikoViewDecorator(mediator) { rect ->
                mediator.sceneBoundsInPx?.let { sceneBounds ->
                    onDrawRectChange(rect.translate(-sceneBounds.topLeft))
                }
            }
        )
```

Reply:
Your code snipped won't work in this (`SwingComposeSceneLayer.desktop.kt`) implementation. I've got your point - I'll make it more clear, but in different way